### ISSUE TITLE
Address accessibility issues on registration and other pages

### ISSUE THREAD

**Issue by alexweissman** [userfrosting/UserFrosting#865]
As per my consult with @vavroom:

The WAVE web-based tool reveals a [number of accessibility issues](http://wave.webaim.org/report#/demo.userfrosting.com). There are a number of issues that may be straight-up invalid HTML as well.

- There appear to be some duplicate labels
- Placeholders in form fields are difficult to read for some due to low contrast, and they disappear once you start typing something. We should go back to fixed labels.
- Labels should be one-to-one with fields. For field groups, we should use a fieldset+legend instead.
- Avoid fixed-height fields. These can cause text to become cut off with browser- or OS- controlled large font sizes.
- Swap the username field and suggest button so the presentation makes more sense when people use screenreaders such as NVDA (it is unclear what will be suggested until you get to the next focused element, the field itself)
- Use the `aria-live` attribute to announce when a field has been programmatically altered (for example, auto-suggested usernames)
- jqueryvalidation doesn't give any feedback to screenreaders when the validation state of a field has changed

**Comment by vavroom** [comment]
I'm totally swamped right now, but as you progress, please feel free to ping me with questions re implementation.
